app/controllers/registrations_controller.rb

```diff
@@ -822,19 +823,69 @@ def create
   end
 
   def create_paypal_order
-    return head :forbidden if Rails.env.production?
+    return head :forbidden if PaypalInterface.paypal_disabled?
 
     @registration = registration_from_params
-    render json: PaypalInterface.create_order(@registration)
+    render json: PaypalInterface.create_order(@registration, params[:total_charge])
   end
 
   def capture_paypal_payment
-    return head :forbidden if Rails.env.production?
+    return head :forbidden if PaypalInterface.paypal_disabled?
 
     @registration = registration_from_params
     @competition = @registration.competition
     order_id = params[:order_id]
 
-    render json: PaypalInterface.capture_payment(@competition, order_id)
+    response = PaypalInterface.capture_payment(@competition, order_id)
+    if response['status'] == 'COMPLETED'
+
+      # TODO: Handle the case where there are multiple captures for a payment
+      # 1) Multiple installments
+      # 2) Some failed, some succeeded
+
+      amount_details = response['purchase_units'][0]['payments']['captures'][0]['amount']
+      currency_code = amount_details['currency_code']
+      amount = PaypalRecord.amount_in_cents(amount_details["value"], currency_code)
+      record = PaypalRecord.find_by(record_id: response["id"])
+
+      # Create a Capture object and link it to the PaypalRecord
+      # NOTE: This assumes there is only ONE capture per order - not a valid long-term assumption
+      capture_from_response = response['purchase_units'][0]['payments']['captures'][0]
```

Comment:
Nit: Rename this to "capture" (but this whole implementation will probably be changed when add in support for multiple captures)

Reply:
We "kinda" already have support for multiple captures (at least structurally). If you mentally map "Paypal Captures" to "Stripe charges" and you look at how the existing "update_status_and_charges" method in Stripe works, you will see that we can simply loop over the charges and create a `RegistrationPayment` for each one.

Reply:
Yeah this is just me doing a thin implementation for the proof of concept - had planned to add that as part of an upcoming edge cases/better implementation PR once we've fleshed out the rest of the modelling

Reply:
(Feel free to reopen conversation if you want me to include that functionality in this PR)